custom_components/tryfi/pytryfi/__init__.py

```diff
@@ -1,106 +1,187 @@
+                        #get behavior stats if available (Series 3+ only)
+                        try:
+                            p.updateBehaviorStats(self._session)
+                        except Exception as e:
+                            LOGGER.debug(f"Behavior stats not available for {p._name} - likely not a Series 3+ collar: {e}")
+                        LOGGER.debug(f"Adding Pet: {p._name} with Device: {p._device._deviceId}")
+                        self._pets.append(p)
+                    else:
+                        LOGGER.warning(f"Pet {pet['name']} - {pet['id']} has no collar. Ignoring Pet!")
+                h = h + 1
+
+            baseListJSON = query.getBaseList(self._session)
+            h = 0
+            for house in baseListJSON:
+                for base in baseListJSON[h]['household']['bases']:
+                    b = FiBase(base['baseId'])
+                    b.setBaseDetailsJSON(base)
+                    LOGGER.debug(f"Adding Base: {b._name} Online: {b._online}")
+                    self._bases.append(b)
+                h = h + 1
+        except Exception as e:
+            capture_exception(e)
 
     def __str__(self):
         instString = f"Username: {self.username}"
         baseString = ""
         petString = ""
         for b in self.bases:
             baseString = baseString + f"{b}"
-        for p in self._pets:
+        for p in self.pets:
             petString = petString + f"{p}"
         return f"TryFi Instance - {instString}\n Pets in Home:\n {petString}\n Bases In Home:\n {baseString}"
 
+    #set the headers for the session
+    def setHeaders(self):
+        self.session.headers = HEADER
+
     #refresh pet details for all pets
     def updatePets(self):
-        for pet in self._pets:
-            pet.updateAllDetails(self._session)
+        try:
+            petListJSON = query.getPetList(self._session)
+            updatedPets = []
+            for house in petListJSON:
+                for pet in house['household']['pets']:
+                    p = FiPet(pet['id'])
+                    p.setPetDetailsJSON(pet)
+                    #get the current location and set it
+                    pLocJSON = query.getCurrentPetLocation(self._session,p._petId)
+                    p.setCurrentLocation(pLocJSON)
+                    #get the daily, weekly and monthly stats and set
+                    pStatsJSON = query.getCurrentPetStats(self._session,p._petId)
+                    p.setStats(pStatsJSON['dailyStat'],pStatsJSON['weeklyStat'],pStatsJSON['monthlyStat'])
+                    #get the daily, weekly and monthly rest stats and set
+                    pRestStatsJSON = query.getCurrentPetRestStats(self._session,p._petId)
+                    p.setRestStats(pRestStatsJSON['dailyStat'],pRestStatsJSON['weeklyStat'],pRestStatsJSON['monthlyStat'])
+                    #get behavior stats if available (Series 3+ only)
```

Comment:
This removes the GraphQL call batching. In the previous version, I made a single GraphQL call to get all the data necessary for the pet. Is that intentional?